biom/table.py

```diff
@@ -611,6 +639,85 @@ def add_group_metadata(self, group_md, axis='sample'):
         else:
             raise UnknownAxisError(axis)
 
+    def del_metadata(self, keys=None, axis=None, inplace=False):
```

Comment:
> Can you add a unit test that uses all default parameters?

Can you also address this comment? Specifically, testing that `table.del_metadata()` works as expected (it used to error).